### PR TITLE
Add new module openms/idripper

### LINKED ISSUE #4640: new module: openms/idripper
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by jonasscheid**
## PR checklist

Closes #4640  

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/openms/idripper/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    process "OPENMS_IDRIPPER"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "openms"
+    tag "openms/idmerger"
+    tag "openms/idripper"
+
+    test("proteomics - split - idxml") {
+        setup {
```
Comment (sateeshperi) [inline modules/nf-core/openms/idripper/tests/main.nf.test:14]: if a setup block is to be used for multiple tests....you can move it beyond any test scope and place it above so it becomes global....read more about global setup [here](https://www.nf-test.com/docs/testcases/setup/)

File: modules/nf-core/openms/idripper/meta.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+name: "openms_idripper"
```
Comment (sateeshperi) [inline modules/nf-core/openms/idripper/meta.yml:1]:
```suggestion
name: openms_idripper
```

File: modules/nf-core/openms/idripper/meta.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+description: Split a merged identification file into their originating identification files
+keywords:
+  - split
+  - idXML
+  - openms
+  - proteomics
+tools:
+  - "openms":
```
Comment (sateeshperi) [inline modules/nf-core/openms/idripper/meta.yml:9]:
```suggestion
  - openms:
```

File: modules/nf-core/openms/idripper/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                """
+                input[0] = OPENMS_IDMERGER.out.idxml
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
```
Comment (sateeshperi) [inline modules/nf-core/openms/idripper/tests/main.nf.test:82]:
```suggestion
                { assert snapshot(process.out.idxmls.collect { file(it[1]).getName() } +
                                process.out.versions).match() }
```

Comment (sateeshperi) [reply modules/nf-core/openms/idripper/tests/main.nf.test:82]: this way we capture just the file name for stub in snapshot and not an empty file

**Review by sateeshperi — CHANGES_REQUESTED** [review]

File: modules/nf-core/openms/idripper/tests/main.nf.test (hunk 4 above)
Comment (mashehu) [reply modules/nf-core/openms/idripper/tests/main.nf.test:82]: You can actually capture everything, nf-core tools ignores empty md5sums for stubs.

**Comment by sateeshperi** [comment]
and the singularity versions are differing....do you have any bandwidth to explore why ?

File: modules/nf-core/openms/idripper/tests/main.nf.test (hunk 1 above)
Comment (jonasscheid) [reply modules/nf-core/openms/idripper/tests/main.nf.test]: Thanks for the hint! Makes way more sense

**Comment by jonasscheid** [comment]
> and the singularity versions are differing....do you have any bandwidth to explore why ?

I think it had to do with the order of files being merged.. I added a config for the merger that enforces the order of the merge

**Review by sateeshperi — APPROVED** [review]
gr8 job!

**Pull request merged**